cmd/native-relayer/cli/start.go

```diff
@@ -0,0 +1,147 @@
+			os.Exit(1)
+		}
+		logLvl, err := zerolog.ParseLevel(lvl)
+		if err != nil {
+			log.Error().Err(err).Msg("Error parsing log level")
+			os.Exit(1)
+		}
+		env.InitLogger(logLvl)
+		configFile, err := cmd.Root().PersistentFlags().GetString("config")
+		if err != nil {
+			log.Error().Err(err).Msg("Error getting config file path")
+			os.Exit(1)
+		}
+
+		config, err := loadConfig(configFile)
+		if err != nil {
+			log.Error().Err(err).Msg("Error loading config")
+			os.Exit(1)
+		}
+		db, err := dal.NewDB(config.DB.File)
+		if err != nil {
+			log.Error().Err(err).Msg("Error creating database")
+			os.Exit(1)
+		}
+		if err := db.InitDB(); err != nil {
+			log.Error().Err(err).Msg("Error initializing database")
+			os.Exit(1)
+		}
+		suiClient := sui.NewSuiClient(config.Ika.RPC).(*sui.Client)
+		if suiClient == nil {
+			log.Error().Err(err).Msg("Error creating Sui client")
+			os.Exit(1)
+		}
+		signer, err := signer.NewSignertWithMnemonic(config.Ika.SignerMnemonic)
+		if err != nil {
+			log.Error().Err(err).Msg("Error creating signer with mnemonic")
+			os.Exit(1)
+		}
+		ikaClient, err := ika.NewClient(
+			suiClient,
+			signer,
+			ika.SuiCtrCall{
+				Package:  config.Ika.NativeLcPackage,
+				Module:   config.Ika.NativeLcModule,
+				Function: config.Ika.NativeLcFunction,
+			},
+			ika.SuiCtrCall{
+				Package:  config.Ika.NativeLcPackage,
+				Module:   config.Ika.NativeLcModule,
+				Function: config.Ika.NativeLcFunction,
+			},
+			config.Ika.GasAcc,
+			config.Ika.GasBudget,
+		)
+		if err != nil {
+			log.Error().Err(err).Msg("Error creating IKA client")
+			os.Exit(1)
+		}
+		btcProcessor, err := ika2btc.NewProcessor(
+			rpcclient.ConnConfig{
+				Host:         config.Btc.RPCHost,
+				User:         config.Btc.RPCUser,
+				Pass:         config.Btc.RPCPass,
+				HTTPPostMode: config.Btc.HTTPPostMode,
+				DisableTLS:   config.Btc.DisableTLS,
+				Params:       config.Btc.Network,
+			},
+			config.Btc.ConfirmationThreshold,
+			db,
+		)
+		if err != nil {
+			log.Error().Err(err).Msg("Error creating Bitcoin processor")
+			os.Exit(1)
+		}
+		nativeProcessor := native2ika.NewProcessor(ikaClient, db)
+		// TODO: replace with the the real endpoint  once available
+		fetcher, err := native.NewMockAPISignRequestFetcher()
+		if err != nil {
+			log.Error().Err(err).Msg("Error creating SignReq fetcher ")
+			os.Exit(1)
+		}
+		relayer, err := nbtc.NewRelayer(
+			nbtc.RelayerConfig{
+				ProcessTxsInterval:   config.Relayer.ProcessTxsInterval,
+				ConfirmTxsInterval:   config.Relayer.ConfirmTxsInterval,
+				SignReqFetchInterval: config.Relayer.SignReqFetchInterval,
+				SignReqFetchFrom:     config.Relayer.SignReqFetchFrom,
+				SignReqFetchLimit:    config.Relayer.SignReqFetchLimit,
+			},
+			db,
+			nativeProcessor,
+			btcProcessor,
+			fetcher,
+		)
+		if err != nil {
+			log.Error().Err(err).Msg("Error creating relayer")
+			os.Exit(1)
+		}
+		var wg sync.WaitGroup
+		wg.Add(1)
```

Comment:
Why do we need it?

Reply:
We need it to ensure the relayer actually stops before displaying `realyer stopped` and exiting

Reply:
I see. Could you add a comment to explain it?

Reply:
Done